addons/website/models/res_lang.py

```diff
@@ -14,20 +15,23 @@ def write(self, vals):
                 raise UserError(_("Cannot deactivate a language that is currently used on a website."))
         return super(Lang, self).write(vals)
 
-    @api.model
     @tools.ormcache_context(keys=("website_id",))
-    def get_frontend_langs(self):
+    def _get_frontend(self) -> LangDataDict:
+        """ Return the available languages for current request
+        :return: LangDataDict({code: LangData})
+        """
         if request and getattr(request, 'is_frontend', True):
-            langs = self.env['website'].get_current_website().language_ids.get_sorted()
+            langs_ids = set(self.env['website'].get_current_website().language_ids.ids)
```

Comment:
The set is not necessary, it is just my habit to add set if I have to use `in` and don't care about order. But since the method is orm cached, the performance is not importatnt, I will remove it as you wish.